src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -264,10 +311,12 @@ public Response downloadZip(JsonObject content,
             Result result = controller.execute();
             if (result instanceof Failed)
             {
-               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(result.getMessage()).build();
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(getMessage(result)).build();
```

Comment:
but figure that'd be a longer term thing - was just going for a simple-ish change in one PR to get most of the benefit of queries & more flexible zip handling